test/integration/testStructuredType.js

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const async = require('async');
+const GlobalConfig = require('./../../lib/global_config');
+const snowflake = require('./../../lib/snowflake');
+const testUtil = require('./testUtil');
+const sharedStatement = require('./sharedStatements');
+const assert = require('assert');
+
+describe('Test Structured types', function () {
+  let connection;
+
+  before(function (done) {
+    connection = testUtil.createConnection({ 'proxyHost': '127.0.0.1', 'proxyPort': 8080 });
+    async.series([
+      function (callback) {
+        snowflake.configure({ 'insecureConnect': true });
+        GlobalConfig.setInsecureConnect(true);
+        testUtil.connect(connection, callback);
+      }],
+    done
+    );
+  });
+
+  describe('test object', function () {
+    it('test simple object', function (done) {
+      const selectObject = 'select {\'string\':\'a\'}::OBJECT(string VARCHAR) as result';
+
+      async.series([
+        function (callback) {
+          testUtil.executeCmd(connection, 'alter session set ENABLE_STRUCTURED_TYPES_IN_CLIENT_RESPONSE = true', callback);
+        },
+        function (callback) {
+          testUtil.executeCmd(connection, 'alter session set IGNORE_CLIENT_VESRION_IN_STRUCTURED_TYPES_RESPONSE = true', callback);
+        },
+        function (callback) {
+          testUtil.executeQueryAndVerify(
+            connection,
+            selectObject,
+            [{ RESULT: { 'string': 'a' } }],
+            callback
+          );
+        }],
+      done
+      );
+    });
+
+    it('test timestamp ltz', function (done) {
+      const selectObject = 'select {' +
+        '\'timestamp_ltz\': \'2021-12-22 09:43:44\'::TIMESTAMP_LTZ' +
+        '}' +
+        '::OBJECT(timestamp_ltz TIMESTAMP_LTZ) As RESULT';
+      async.series([
+        function (callback) {
+          const executeOptions = {};
+          executeOptions.sqlText = selectObject;
+          executeOptions.complete = function (err, stmt) {
+            assert.ok(!err, JSON.stringify(err));
+            let rowCount = 0;
+            const stream = stmt.streamRows();
+
+            stream.on('readable', function () {
+              let row;
+              while ((row = stream.read()) !== null) {
+                const narmalizedRow = {};
+                const expected = {};
+                Object.keys(row).forEach((key) => {
+                  narmalizedRow[key] = testUtil.normalizeRowObject(row[key]);
+                });
+                expected.timestamp_ltz = '2021-12-22 09:43:44.000 -0800';
```

Comment:
Why do we have a timezone here? Is it something that is returned to the customer?